lib/settings_page.dart

```diff
@@ -18,102 +24,246 @@ class _MySettingsPageState extends State<MySettingsPage> {
   //State and Behavior
   bool _showProfile;
   bool _showSettings;
+  bool _showPreferences;
+  UserAuthentication auth;
 
   @override
   void initState() {
     _showProfile = false;
-    _showSettings = false;
+    _showSettings = true;
+    _showPreferences = false;
     super.initState();
   }
 
   void showProfile() {
-    setState(
-      () {
+    setState(() {
+      _showProfile = !_showProfile;
+      if (_showSettings) {
+        _showSettings = !_showSettings;
+      }
+    });
+  }
+
+  void showSettings() {
+    setState(() {
+      _showSettings = !_showSettings;
+      if (_showProfile) {
         _showProfile = !_showProfile;
-        if (_showSettings) {
-          _showSettings = !_showSettings;
-        }
-      },
+      }
+    });
+  }
+
+  void showPreferences() {
+    setState(() {
+      _showPreferences = !_showPreferences;
+      if (_showPreferences) {
+        PreferenceScreen();
+      }
+    });
+  }
+
+  Column buildProfileButton() {
```

Comment:
we can reuse this function instead of creating redundant code. You can pass as trying that get's passed to the Text class. We can rename the function to `buildButton(String columnName, showProfile())` 

here are resources on how we can create functions that accept functions as a parameter: https://alvinalexander.com/flutter/dart-how-pass-function-into-function-method-parameter/

Reply:
Deng good call here, missed this! Will fix

Reply:
I've tried to reuse this function, but because it's a dynamic button, it will break functionality when trying to extract the function.

Reply:
aww ok